src/common/HttpRequest.cc

```diff
@@ -0,0 +1,156 @@
+/* -------------------------------------------------------------------------- */
+/* Copyright 2002-2023, OpenNebula Project, OpenNebula Systems                */
+/*                                                                            */
+/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
+/* not use this file except in compliance with the License. You may obtain    */
+/* a copy of the License at                                                   */
+/*                                                                            */
+/* http://www.apache.org/licenses/LICENSE-2.0                                 */
+/*                                                                            */
+/* Unless required by applicable law or agreed to in writing, software        */
+/* distributed under the License is distributed on an "AS IS" BASIS,          */
+/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
+/* See the License for the specific language governing permissions and        */
+/* limitations under the License.                                             */
+/* -------------------------------------------------------------------------- */
+
+#include "HttpRequest.h"
+#include <iostream>
+
+using namespace std;
+
+
+HttpRequest::HttpRequest()
+{
+    curl_global_init(CURL_GLOBAL_ALL);
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+HttpRequest::~HttpRequest()
+{
+    curl_global_cleanup();
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+int HttpRequest::post_json(const std::string& url, const std::string& data, std::string& response)
+{
+    auto curl = curl_easy_init();
+
+    if (!curl) return -1;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that in these codes. `Multiline blocks` are missing. It is considered best practice to enclose all `multiline blocks` including if else blocks, loops, and function bodies within curly braces. To mitigate this issue, it serves to clearly define the scope of the block and avoid logic errors by grouping all related statements together visually.